clang/lib/CodeGen/CGGPUBuiltin.cpp

```diff
@@ -178,17 +181,29 @@ RValue CodeGenFunction::EmitNVPTXDevicePrintfCallExpr(const CallExpr *E) {
       E, this, GetVprintfDeclaration(CGM.getModule()), false);
 }
 
+// Deterimines if an argument is a string
+static bool isString(const clang::Type *argXTy) {
```

Comment:
This could be called "MayBeString()" at best. It's not necessary that a char* argument type is a C-style string.